packages/typespec-azure-resource-manager/lib/Legacy/decorator.tsp

```diff
@@ -7,3 +7,9 @@ namespace Azure.ResourceManager.Legacy;
  * but need to be identified as such.
  */
 extern dec customAzureResource(target: Model);
+
+/**
+ * Specify an external reference that should be used when emitting this type.
+ * @param jsonRef - External reference(e.g. "../../common.json#/definitions/Foo")
+ */
+extern dec commonTypesRef(entity: Model | ModelProperty, jsonRef: valueof string);
```

Comment:
I cannot think of a scenario that could have `ModelProperty` as target. Do we have such cases?

Reply:
if we broaden to @externTypeRef then it is conceivable a property could have a direct ref to a type in a different file rather than having to have a separate model definition. Also the model itself can be a spread type. There are a few examples in ARM common-type ref.